### PR TITLE
feat: initial buildpacks for C++

### PR DESCRIPTION
**Pull request by coryan**
Initial support to build functions using build packs.

### CODE REVIEW

File: pack/buildpack/bin/build [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+cat > "${layers}/vcpkg.toml" <<EOL
+build = true
+cache = true
+launch = false
+EOL
+
+echo "-----> Setup build directory"
+cp -r /layers/cpp/gcf/cmake /var/tmp/source
+cp -r /workspace /var/tmp/source/application
+if [ -r /workspace/vcpkg.json ]; then cp /workspace/vcpkg.json /var/tmp/source; fi
```
Comment (devjgm) [inline pack/buildpack/bin/build:38]: Instead of an `if` one-liner, consider `test -r <file> && cp <file> /var/tmp/source/`

**Review by devjgm — APPROVED** [review]
LGTM, but I know almost nothing about build packs. Is there someone who understands build packs who you'd like to look at this too?

File: pack/buildpack/bin/build (hunk 1 above)
Comment (coryan) [reply pack/buildpack/bin/build]: Fixed.

**Comment by coryan** [comment]
@grant can you double check the buildpack and/or recommend somebody to take a look?  I have a followup PR that actually uses the buildpack to create an image and run them.

**Comment by grant** [comment]
@squee1945 can you check this Buildpack?

**Comment by squee1945** [comment]
This looks to follow the buildpacks.io spec, but to use it in our GCP buildpacks builder images (including GAE and GCF), it will need to be implemented in Go, over in https://github.com/GoogleCloudPlatform/buildpacks (of course, this code can serve as a great roadmap for that implementation).

**Comment by coryan** [comment]
Thanks, we can revisit the reimplementation in golang later.

**Pull request merged**